actors/builtin/market/market_state.go

```diff
@@ -346,7 +475,7 @@ func (st *State) slashBalance(et, lt *adt.BalanceTable, addr addr.Address, amoun
 ////////////////////////////////////////////////////////////////////////////////
 
 func (st *State) mustGetDeal(rt Runtime, dealID abi.DealID) *DealProposal {
-	proposals, err := AsDealProposalArray(adt.AsStore(rt), st.Proposals)
+	proposals, err := st.CachedProposals(adt.AsStore(rt))
```

Comment:
This pattern would be repeated throughout state if we adopted this approach.  I held off from doing this with updatePendingDealState and processDealInitTimedOut because the cache read would introduce err values to those methods.